odoo/addons/test_new_api/tests/test_properties.py

```diff
@@ -1728,6 +1733,127 @@ def test_properties_server_action_path_traversal(self):
             "The path contained by the field 'Field to Update Path' contains a non-relational field (Properties) that is not the last field in the path. You can't traverse non-relational fields (even in the quantum realm). Make sure only the last field in the path is non-relational."
         )
 
+    def test_getitem_property(self):
+        # read a property that exist nowhere
+        self.assertEqual(self.message_3['attributes']['____'], None)
+
+        # read a property that exists, but on the wrong record
+        self.assertEqual(self.message_3['attributes']['moderator_partner_id'], None)
```

Comment:
That's a very bad idea.  If a key doesn't exist in a `dict`, accessing it raises a `KeyError` exception.  That's what you should get.